undefined reference to strsep using MinGW-W64 GCC 8.1

Hello,
I'm trying to build the docs, but with the latest additions to GIT, the docs target fails to build.

```
[ 76%] Linking C executable scan_examples.exe
[ 76%] [ 76%] Built target make_search_index
Built target make_index
[ 76%] Building C object docs/CMakeFiles/insert_timestamp.dir/scripts/trex.c.obj
[ 76%] Generating texi/changes-5.1.texi
[ 77%] Building C object docs/CMakeFiles/make_doc.dir/scripts/trex.c.obj
[ 77%] Built target make_protos
[ 77%] Generating texi/changes-5.2.texi
[ 77%] Built target make_html_refs
CMakeFiles\scan_examples.dir/objects.a(scan_examples.c.obj):scan_examples.c:(.text.startup+0xfd): undefined reference to `strsep'
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [docs\CMakeFiles\scan_examples.dir\build.make:134: docs/scan_examples.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:596: docs/CMakeFiles/scan_examples.dir/all] Error 2
mingw32-make[1]: *** Waiting for unfinished jobs....
[ 77%] Built target docs
[ 77%] Linking C executable insert_timestamp.exe
[ 77%] Linking C executable make_doc.exe
CMakeFiles\make_doc.dir/objects.a(make_single.c.obj):make_single.c:(.text+0x1df): undefined reference to `strsep'
CMakeFiles\make_doc.dir/objects.a(make_single.c.obj):make_single.c:(.text+0x1f7): undefined reference to `strsep'
collect2.exe: error: ld returned 1 exit status
[ 77%] Built target insert_timestamp
mingw32-make[2]: *** [docs\CMakeFiles\make_doc.dir\build.make:166: docs/make_doc.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:633: docs/CMakeFiles/make_doc.dir/all] Error 2
mingw32-make: *** [Makefile:129: all] Error 2

e:\usr\libs\Allegro52X\build>mingw32-make
[ 59%] Built target allegro_monolith
[ 61%] Built target docs
[ 63%] Built target make_search_index
[ 65%] Built target make_protos
[ 67%] Built target make_index
[ 69%] Built target make_html_refs
[ 70%] Linking C executable scan_examples.exe
CMakeFiles\scan_examples.dir/objects.a(scan_examples.c.obj):scan_examples.c:(.text.startup+0xfd): undefined reference to `strsep'
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [docs\CMakeFiles\scan_examples.dir\build.make:134: docs/scan_examples.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:596: docs/CMakeFiles/scan_examples.dir/all] Error 2
mingw32-make: *** [Makefile:129: all] Error 2
```

--- a/docs/scripts/make_single.c
+++ b/docs/scripts/make_single.c
@@ -154,15 +154,18 @@ static void insert_examples(void) {
          char* pitem = strcpy(items, exs);
          d_print("Examples:\n");
          char* item;
-         while ((item = strsep(&pitem, " ")) != NULL) {
+         for (item = strtok(pitem, " "); item; item = strtok(NULL, " ")) {
             dstr buffer;
-            char* filename = strsep(&item, ":");
-            if (item) {
+            char* colon = strchr(item, ':');
+            if (colon) {
+               char* filename = item;
+               char* line = colon + 1;
+               *colon = '\0';
                dstr base;
                d_basename(filename, NULL, base);
                d_printf("* [%s](%s)\n",
                         base,
-                        example_source(buffer, filename, item));
+                        example_source(buffer, filename, line));
             }
          }
          d_print("");
@@ -173,6 +176,7 @@ static void insert_examples(void) {
 
 /* Local Variables: */
 /* c-basic-offset: 3 */
+/* indent-tabs-mode: nil */
 /* End: */
 
 /* vim: set sts=3 sw=3 et: */

--- a/docs/scripts/scan_examples.c
+++ b/docs/scripts/scan_examples.c
@@ -175,12 +175,14 @@ void load_api_list(void)
    while (d_getline(line)) {
          int i;
          bool found = false;
-         char *ptr = line;
-         strsep(&ptr, ":");
-         for (i = apis.count - 1; i >=0; --i) {
-            if (strcmp(line, apis.items[i]) == 0) {
+         char *ptr = strchr(line, ':');
+         if (ptr) {
+            *ptr = '\0';
+            for (i = apis.count - 1; i >=0; --i) {
+               if (strcmp(line, apis.items[i]) == 0) {
                   found = true;
                   break;
+               }
             }
          }
          if (!found) {
@@ -232,5 +234,6 @@ char **responsefile(int *pargc, char **argv)
 
 /* Local Variables: */
 /* c-basic-offset: 3 */
+/* indent-tabs-mode: nil */
 /* End: */
 /* vim: set sts=3 sw=3 et: */

